pyomo/contrib/solver/solution.py

```diff
@@ -0,0 +1,279 @@
+# May turn into documentation someday
+class SolutionLoader(SolutionLoaderBase):
+    def __init__(
+        self,
+        primals: Optional[MutableMapping],
+        duals: Optional[MutableMapping],
+        reduced_costs: Optional[MutableMapping],
+    ):
+        """
+        Parameters
+        ----------
+        primals: dict
+            maps id(Var) to (var, value)
+        duals: dict
+            maps Constraint to dual value
+        reduced_costs: dict
+            maps id(Var) to (var, reduced_cost)
+        """
+        self._primals = primals
+        self._duals = duals
+        self._reduced_costs = reduced_costs
+
+    def get_primals(
+        self, vars_to_load: Optional[Sequence[_GeneralVarData]] = None
+    ) -> Mapping[_GeneralVarData, float]:
+        if self._primals is None:
+            raise RuntimeError(
+                'Solution loader does not currently have a valid solution. Please '
+                'check the termination condition.'
+            )
+        if vars_to_load is None:
+            return ComponentMap(self._primals.values())
+        else:
+            primals = ComponentMap()
+            for v in vars_to_load:
+                primals[v] = self._primals[id(v)][1]
+            return primals
+
+    def get_duals(
+        self, cons_to_load: Optional[Sequence[_GeneralConstraintData]] = None
+    ) -> Dict[_GeneralConstraintData, float]:
+        if self._duals is None:
+            raise RuntimeError(
+                'Solution loader does not currently have valid duals. Please '
+                'check the termination condition and ensure the solver returns duals '
+                'for the given problem type.'
+            )
+        if cons_to_load is None:
+            duals = dict(self._duals)
+        else:
+            duals = {}
+            for c in cons_to_load:
+                duals[c] = self._duals[c]
+        return duals
+
+    def get_reduced_costs(
+        self, vars_to_load: Optional[Sequence[_GeneralVarData]] = None
+    ) -> Mapping[_GeneralVarData, float]:
+        if self._reduced_costs is None:
+            raise RuntimeError(
+                'Solution loader does not currently have valid reduced costs. Please '
+                'check the termination condition and ensure the solver returns reduced '
+                'costs for the given problem type.'
+            )
+        if vars_to_load is None:
+            rc = ComponentMap(self._reduced_costs.values())
+        else:
+            rc = ComponentMap()
+            for v in vars_to_load:
+                rc[v] = self._reduced_costs[id(v)][1]
+        return rc
+
+
+class SolSolutionLoader(SolutionLoaderBase):
+    def __init__(self, sol_data: SolFileData, nl_info: NLWriterInfo) -> None:
+        self._sol_data = sol_data
+        self._nl_info = nl_info
+
+    def load_vars(
+        self, vars_to_load: Optional[Sequence[_GeneralVarData]] = None
+    ) -> NoReturn:
```

Comment:
Because you were concerned about performance. We had an implementation that did not ignore `vars_to_load`, and then decided (for performance reasons) `vars_to_load` is the minimum set of variable values to load.